### ISSUE TITLE
Flutter Couldn't find meta-data for provider with authority webview_plugin.WebviewManager$3.onShowFileChooser

### ISSUE THREAD

**Issue by arashAndroid** [fluttercommunity/flutter_webview_plugin#592]
I am trying to use flutter_webview_plugin to show content of a url. In the url I have a button that opens a file explorer to choose a file.This works fine in browsers but in flutter_webview_plugin it gives this error:

```
W/System.err(15213): java.lang.IllegalArgumentException: Couldn't find meta-data for provider with authority com.example.booshehr.fileprovider
W/System.err(15213):    at androidx.core.content.FileProvider.parsePathStrategy(FileProvider.java:606)
W/System.err(15213):    at androidx.core.content.FileProvider.getPathStrategy(FileProvider.java:579)
W/System.err(15213):    at androidx.core.content.FileProvider.getUriForFile(FileProvider.java:417)
W/System.err(15213):    at com.flutter_webview_plugin.WebviewManager.getOutputFilename(WebviewManager.java:272)
W/System.err(15213):    at com.flutter_webview_plugin.WebviewManager.access$800(WebviewManager.java:42)
W/System.err(15213):    at com.flutter_webview_plugin.WebviewManager$3.onShowFileChooser(WebviewManager.java:211)
W/System.err(15213):    at In.a(PG:145)
W/System.err(15213):    at uE0.runFileChooser(PG:2)
W/System.err(15213):    at android.os.MessageQueue.nativePollOnce(Native Method)
W/System.err(15213):    at android.os.MessageQueue.next(MessageQueue.java:325)
W/System.err(15213):    at android.os.Looper.loop(Looper.java:142)
W/System.err(15213):    at android.app.ActivityThread.main(ActivityThread.java:6710)
W/System.err(15213):    at java.lang.reflect.Method.invoke(Native Method)
W/System.err(15213):    at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
W/System.err(15213):    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:770)
F/chromium(15213): [FATAL:jni_android.cc(249)] Please include Java exception stack in crash report
F/libc    (15213): Fatal signal 5 (SIGTRAP), code 1 in tid 15213 (xample.booshehr)
*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
Build fingerprint: 'lge/elsa_global_com/elsa:8.0.0/OPR1.170623.032/18325232422a1:user/release-keys'
Revision: '12'
ABI: 'arm64'
pid: 15213, tid: 15213, name: xample.booshehr  >>> com.example.booshehr <<<
signal 5 (SIGTRAP), code 1 (TRAP_BRKPT), fault addr 0x714216eb04
    x0   0000000000000000  x1   0000000000000081  x2   000000007fffffff  x3   0000000000000000
    x4   0000000000000000  x5   0000000000000000  x6   0000000000000000  x7   7f7f7f7f7fffff7f
    x8   0000000000000000  x9   0000000000000000  x10  0000000000000001  x11  0000000000000000
    x12  0000007134d6ff00  x13  0000000000000000  x14  0000000000000000  x15  000000716228c000
    x16  000000715dfd2208  x17  000000715df25c50  x18  000000006fba3d30  x19  0000007fd17f5c70
    x20  0000007fd17f5c78  x21  0000007fd17f5c80  x22  000000000000004f  x23  000000715dfd4808
    x24  0000007143e3f000  x25  0000007fd17f5e70  x26  0000000000000001  x27  0000007162612a48
    x28  0000007162612a48  x29  0000007fd17f5c60  x30  000000714216e9c4
    sp   0000007fd17f57b0  pc   000000714216eb04  pstate 0000000060000000
backtrace:
    #00 pc 0000000001e9fb04  /data/app/com.android.chrome-HU8bsIOJpUMqks882ELhSA==/base.apk (offset 0xc19000)
    #01 pc 0000000001e9f9c0  /data/app/com.android.chrome-HU8bsIOJpUMqks882ELhSA==/base.apk (offset 0xc19000)

```
The button open a bottom widget that has a list of actions like camera, files and ...  in chrome.
**flutter doctor -v**
```
[√] Flutter (Channel stable, v1.9.1+hotfix.4, on Microsoft Windows [Version 6.3.9600], locale en-US)
    • Flutter version 1.9.1+hotfix.4 at D:\flutter
    • Framework revision cc949a8e8b (6 weeks ago), 2019-09-27 15:04:59 -0700
    • Engine revision b863200c37
    • Dart version 2.5.0

[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at E:\Android\androidSDK
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = E:\Android\androidSDK
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b02)
    • All Android licenses accepted.

[√] Android Studio (version 3.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 29.0.1
    • Dart plugin version 173.4700
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b02)

[√] VS Code (version 1.39.2)
    • VS Code at C:\Users\Rubcik\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.6.0

[√] Connected device (1 available)
    • LG H990 • LGH9903994715a • android-arm64 • Android 8.0.0 (API 26)

• No issues found!
```
I have been looking everywhere to find a solution and things that I have tried are :
I have tried adding provider to androidmanifest, still the same error:
```
<provider
                android:name="androidx.core.content.FileProvider"
                android:authorities="${applicationId}.fileProvider"
                android:exported="false"
                android:grantUriPermissions="true"
                tools:replace="android:authorities">
            <meta-data
                    android:name="android.support.FILE_PROVIDER_PATHS"
                    android:resource="@xml/filepaths"
                    tools:replace="android:resource" /> 
</provider>
```
my @xml/filepaths is:
```
<?xml version="1.0" encoding="utf-8"?> 
<paths> 
<files-path name="myapp" path="images/"/>
</paths>
```
Then I tried changing the `android:authorities="${applicationId}.fileProvider"` to `android:authorities="${applicationId}.myOwnFileProvider"` , still the same error.

Please help me on this.

**Comment by littlesalt** [comment]
@arashAndroid Hi, I encountered the same problem, did you solve it?

**Comment by arashAndroid** [comment]
@littlesalt I had to remove 'open_file' package from my app. It solved the error but it cost me a little (can't open files anymore :) ).
the 'open_file' package actually had a solution for this in their package page, but that did not work for me.
So you need to look for a package that does something with files (reading writing ...) and remove it one by one. after you found the package(s) that cause the problem, head to their package page and see if they have solution for it.

**Comment by littlesalt** [comment]
I also had to remove 'open_file' package from my app. and who can instead of "open_file"？Please recommend one！

**Comment by arashAndroid** [comment]
@littlesalt I did not use any other packages. Sorry I don't know other packages. You have to search and test them.

**Comment by littlesalt** [comment]
@arashAndroid OK,thanks!

**Comment by Theusmatag** [comment]
hello guys this problem today I used the inappwebview package: https://pub.dev/packages/flutter_inappwebview/example

the solution was as follows: 

1. Open the `AndroidManifest.xml` file located in the `android/app/src/main` directory of your Flutter project.

2. Check if you have the following declaration inside the `<application>` tag in the `AndroidManifest.xml` file:
   ```xml
   <provider
       android:name="androidx.core.content.FileProvider"
       android:authorities="br.com.delbank.flutter_inappwebview.fileprovider"
       android:exported="false"
       android:grantUriPermissions="true">
       <meta-data
           android:name="android.support.FILE_PROVIDER_PATHS"
           android:resource="@xml/path_provider_paths" />
   </provider>
   ```

4. Verify if you have a file named `path_provider_paths.xml` in the `android/app/src/main/res/xml` directory. If the file doesn't exist, create a new XML file named `path_provider_paths.xml` in that directory.

5. Open the `path_provider_paths.xml` file and add the following code:
   ```xml
   <?xml version="1.0" encoding="utf-8"?>
   <paths xmlns:android="http://schemas.android.com/apk/res/android">
       <external-path name="external_files" path="." />
   </paths>
   ```

6. Save the modified files.

7. Run the application again to see if the error has been resolved.

These steps should help you resolve the "java.lang.IllegalArgumentException: Couldn't find meta-data for provider with authority br.com.delbank.flutter_inappwebview.fileprovider" error related to the file provider in Flutter. Make sure you have entered the correct information in the `AndroidManifest.xml` file and created the `path_provider_paths.xml` file correctly.
